### PR TITLE
at86rf2xx: document operation mode concerns for

### PR DESCRIPTION
**Pull request by miri64**
Documents concerns stated in https://github.com/RIOT-OS/RIOT/pull/4925#issuecomment-190735863

### CODE REVIEW

**Comment by OlegHahm** [comment]
Maybe you can provide a pointer on how this randomness was shown?

**Comment by miri64** [comment]
What exactly do you mean by that? An application?

**Comment by OlegHahm** [comment]
I don't know. How did you show randomness?

**Comment by miri64** [comment]
I ran `tests/driver_at86rf2xx` with the patch at https://github.com/RIOT-OS/RIOT/pull/4925#issuecomment-190735863 and checked if the values were different every time. But if you like I can write something up that provides a histogram for a given sample or something like that.

**Comment by OlegHahm** [comment]
Maybe some of [these tests](https://en.wikipedia.org/wiki/Randomness_tests) could give a hint?

**Comment by OlegHahm** [comment]
I just think that "has been shown" has a similar semantic as "has been proven".

**Comment by miri64** [comment]
> I just think that "has been shown" has a similar semantic as "has been proven".

Is the new wording better suited?

**Comment by OlegHahm** [comment]
I would prefer to be really conservative with the wording here and explicitly add that this feature should not be used as a seed for crypto-relevant randomness.

File: drivers/at86rf2xx/include/at86rf2xx_internal.h [hunk 1]
```diff
@@ -163,10 +163,18 @@ void at86rf2xx_hardware_reset(at86rf2xx_t *dev);
...
-#if defined MODULE_AT86RF233 || defined MODULE_AT86RF231
+#if defined(MODULE_AT86RF233) || defined(MODULE_AT86RF231) || defined(DOXYGEN)
 /**
  * @brief   Read random data from the RNG
  *
+ * @note    According to the data sheet this function only works properly in
+ *          Basic Operation Mode. However, sporadic testing has shown that even
+ *          in Extended Operation Mode this returns random enough data to be
+ *          used as a seed for @ref sys_random for not relevant for
+ *          cryptography.
```
Comment (OlegHahm) [inline drivers/at86rf2xx/include/at86rf2xx_internal.h]: Maybe "used as a seed for @ref sys_random if no cryptographically secure randomness is required." ?

**Comment by miri64** [comment]
I would call that "good enough" for the given use-case

```
$ ent at86rf2xx_nrg
Entropy = 7.989497 bits per byte.

Optimum compression would reduce the size
of this 12805768 byte file by 0 percent.

Chi square distribution for 12805768 samples is 188371.01, and randomly
would exceed this value less than 0.01 percent of the times.

Arithmetic mean value of data bytes is 131.8571 (127.5 = random).
Monte Carlo value for Pi is 3.051011716 (error 2.88 percent).
Serial correlation coefficient is -0.031957 (totally uncorrelated = 0.0).
```

(generated using [this application](https://github.com/miri64/RIOT_playground/tree/master/at86rf2xx_rng))

**Comment by miri64** [comment]
(do we know these numbers for any of the hwnrg devises?)

File: drivers/at86rf2xx/include/at86rf2xx_internal.h (hunk 1 above)
Comment (miri64) [inline drivers/at86rf2xx/include/at86rf2xx_internal.h]: Well momentarily this is the only source of randomness we have on these boards, so it is not an "if" (except you think 0 is then a better seed ;-))

Comment (OlegHahm) [inline drivers/at86rf2xx/include/at86rf2xx_internal.h]: First of all, the current sentence is broken anyway (syntactically).
Second, IMO weak or wrong security is worse than no security.

Comment (miri64) [inline drivers/at86rf2xx/include/at86rf2xx_internal.h]: Fixed.

**Comment by OlegHahm** [comment]
No, and I just heard from a security researcher about completely broken hardware random number generators. However, I think if a vendor states that its random number generator **is** cryptographically secure, you should not be to sure about it, but the vendor does not even state this, we should be very, very careful.

Furthermore, in order to get valid estimation of its entropy, one would probably need to understand how it works and why this particular note exists in the datasheet.

File: drivers/at86rf2xx/include/at86rf2xx_internal.h (hunk 1 above)
Comment (miri64) [inline drivers/at86rf2xx/include/at86rf2xx_internal.h]: > Second, IMO weak or wrong security is worse than no security.

I prefer having randomness (even bad one) over none at all.

**Comment by OlegHahm** [comment]
However, I'm fine with the currently proposed text. :-) ACK

File: drivers/at86rf2xx/include/at86rf2xx_internal.h (hunk 1 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/include/at86rf2xx_internal.h]: I agree - but only for non-cryptographically secure functionality.

**Comment by miri64** [comment]
Squashed and rebased.

**Pull request merged**